proposed/http-client/http-client.md

```diff
@@ -41,12 +41,10 @@ use Psr\Http\Message\ResponseInterface;
 interface ClientInterface
 {
     /**
-     * Sends a PSR-7 request.
-     *
-     * If a request is sent without any prior configuration, an exception MUST NOT be thrown
-     * when a response is recieved, no matter the HTTP status code.
-     *
-     * If a request is sent without any prior configuration, a HTTP client MUST NOT follow redirects.
+     * Sends a PSR-7 request and returns a PSR-7 response. 
+     * 
+     * Every technically correct HTTP response MUST be returned as is, even if it represents a HTTP 
+     * error response or a redirect instruction.
```

Comment:
should we be explicit about the 1xx status? i think we should, and we should mandate that the default configuration does resolve 1xx itself. i guess not many people will write low-level debugging tools with php, and when they do they probably either use sockets directly or a specific client that they can tightly control in the tool.